### PR TITLE
Provide workaround for _GetFocusedTabIndex

### LINKED ISSUE #1039: It is generally possible to have a TerminalControl focused that doesn't exist anymore
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.18898.1000]
Windows Terminal version (if applicable): 

Any other software?
```

# Steps to reproduce
1. Open terminal (Assuming Powershell is the default profile)
2. Open two cmd tabs
3. Use middle click to close those two cmd tabs

# Expected behavior
Two cmd tabs are closed and PowerShell is in focus

# Actual behavior
CMD is still in focus despite it is not shown in the tab bar

Might be related to #767

### LINKED ISSUE #1074: Debug Assertion Failure 'Expression: vector subscript out of range'"
# Environment

```none
Windows build number: Microsoft Windows [Version 10.0.18908.1000]
Windows Terminal version (if applicable): 71e19cd82528d66a0a7867cbed85990cfc1685f1
```

# Steps to reproduce

launch Terminal
create cmd tab
close PowerShell tab via x on tab
type `word^W` into cmd

# Expected behavior

Close the terminal (really, I would like readine completion)

# Actual behavior

Invalid vector access.

Seems that the `_GetFocusedTabIndex` is returning the index unadjusted for the closing of the tab.  I think that `_RemoveTabViewItem` is not readjusting the index. The `shared_ptr` is constructed by indexing into `_tabs` by `_GetFocusedTabIndex`.  That calculates the index by `_tabView.SelectedIndex`  The `_RemoveTabViewItem` does not update the index.  I can not seem to track down where the implementation for the tab bar renderer is handled (I suspect due to the Xaml bits).  Some pointers as to where to start digging would be appreciated.

### LINKED ISSUE #1100: Is there any reason to not use tabView.SelectedIndex()?
https://github.com/microsoft/terminal/blob/880272c7483a3bb6893dbec3725d0eb7375ab78a/src/cascadia/TerminalApp/App.cpp#L632-L639

Upon investigating #1074 & #1038, I was wondering if there is any reason that we are going through all the hassle of finding the correct `tabViewItem` from `vector tabs` instead of just something like
```cpp
    void App::_SetFocusedTabIndex(int tabIndex)
    {
        _tabView.SelectedIndex(tabIndex);
    }
```
I guess the diffrence is that I use the index from `_tabView.Item()` instead, but does `tabs` and `_tabView.Items()` ever get out of sync?

### PR DESCRIPTION
**Pull request by Summon528**
## Summary of the Pull Request
Use tabview.SelectedIndex for setting focus tab

 
## PR Checklist
* [x] Closes #1100, Closes #1039, Closes #1074
* [x] CLA signed. If not, go over [here](https://cla.opensource.microsoft.com/microsoft/Terminal) and sign the CLA
* [ ] Tests added/passed
* [ ] Requires documentation to be updated
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

### CODE REVIEW

File: src/cascadia/TerminalApp/App.cpp [hunk 1]
```diff
@@ -631,11 +631,7 @@ namespace winrt::TerminalApp::implementation
 
     void App::_SetFocusedTabIndex(int tabIndex)
     {
-        auto tab = _tabs.at(tabIndex);
-        _tabView.Dispatcher().RunAsync(CoreDispatcherPriority::Normal, [tab, this](){
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/App.cpp]: You might need to add the Dispatcher.RunAsync(...) code back - I'm not sure this method is always called on the UI thread. 

For example, does pressing Alt+\<number\> to switch to a specific tab work with the change as it is currently?

Comment (Summon528) [reply src/cascadia/TerminalApp/App.cpp]: Uhh, yes it still works :P. Should I add it make nevertheless?

Comment (zadjii-msft) [reply src/cascadia/TerminalApp/App.cpp]: ... Probably yea. Might save someone else in the future :)

Comment (Summon528) [reply src/cascadia/TerminalApp/App.cpp]: Fixed

Comment (Summon528) [reply src/cascadia/TerminalApp/App.cpp]: Fixed

**Comment by Summon528** [comment]
Ohhh bad news. This change breaks the "shift focus to the right after closing tab" behavior. So if I am focusing on tab0 and close tab0, the code will remove tab0, after that the new tab0 is the old tab1. The code will then call _tabview.SelectedIndex(0) but tabview won't change focus to the new tab0.

In general, I think there is a problem upstream with `tabview.SelectedIndex()`, another example is `tabview.SelectedIndex()` sometimes return different result from `_tabView.Items().IndexOf(_tabView.SelectedItem(), x);` If you replace 
```cpp
    int App::_GetFocusedTabIndex() const
    {
        return _tabView.SelectedIndex();
    }
```
to
```cpp
    int App::_GetFocusedTabIndex() const
    {
        uint32_t focusedIndex;
        _tabView.Items().IndexOf(_tabView.SelectedItem(), focusedIndex);
        return focusedIndex;
    }
```
#1039 and #1074 suddenly get fixed.  

I am now unsure what to do with this PR, should we use the workaround version in the two functions or use the more semantically correct version in both functions, or just leave it as it currently is.

**Comment by zadjii-msft** [comment]
It's certainly possible there's an upstream bug - we're using a _super early_ prerelease version of the TabView, so all the bugs haven't really been worked out quite yet.

If you add the change that you say will fix 1039 and 1074, then does that also restore the "shift focus to the right after closing tab" behavior? 

If it _does_, then I'd say throw the fix in this PR, and knock 3 issues out at once. 

Otherwise, I'd probably make a new PR for that other fix, and just close this one, since it doesn't totally resolve the issue. The original issue was "Is there any reason to not use tabView.SelectedIndex()? " after all, and as it turns out, the answer is _"There is a reason"_

**Comment by Summon528** [comment]
Yes, the fix restores the behavior. I have modified this PR for that fix.

**Pull request merged**